plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-button/payment-gateway-button.tsx

```diff
@@ -56,6 +59,12 @@ export const PaymentGatewayButton = ( {
 	};
 
 	const onClick = ( e: React.MouseEvent ) => {
+		if ( isWCPay && needsSetup ) {
+			e.preventDefault();
+			window.location.href = getAdminLink(
```

Comment:
That’s an excellent suggestion, Oleks. I don’t think we need a patch since we can navigate directly to the page. I’ve removed the code and updated the testing instructions accordingly.